server/src/main/java/com/defold/extender/builders/CSharpBuilder.java

```diff
@@ -177,35 +189,56 @@ private static Path getNativePath(String platform) throws IOException {
         Path aotBase = getNativePath(platform);
 
-        linkFlags.add(Paths.get(aotBase.toString(), "libbootstrapperdll.o").toString());
+        ArrayList<String> paths = new ArrayList<>();
 
-        // Note: These libraries are specified with full paths, or the linker will link against the dynamic libraries.
-        // We want to avoid that hassle for now. Let's do that in a step two.
+        paths.add(getObjName(platform, "bootstrapperdll"));
 
         // TODO: Do we need a way to toggle these behaviors on/off?
-        linkFlags.add(Paths.get(aotBase.toString(), "libRuntime.WorkstationGC.a").toString());
-        linkFlags.add(Paths.get(aotBase.toString(), "libeventpipe-enabled.a").toString());
-        linkFlags.add(Paths.get(aotBase.toString(), "libstandalonegc-enabled.a").toString());
+        paths.add(getLibName(platform, "Runtime.WorkstationGC"));
+        paths.add(getLibName(platform, "eventpipe-enabled"));
+        paths.add(getLibName(platform, "standalonegc-enabled"));
 
-        linkFlags.add(Paths.get(aotBase.toString(), "libSystem.Native.a").toString());
-        linkFlags.add(Paths.get(aotBase.toString(), "libSystem.IO.Compression.Native.a").toString());
-        linkFlags.add(Paths.get(aotBase.toString(), "libSystem.Globalization.Native.a").toString());
+        String aotSuffix = "";
+        if (ExtenderUtil.isWindowsTarget(platform))
+            aotSuffix = ".Aot";
+        paths.add(getLibName(platform, "System.IO.Compression.Native" + aotSuffix));
+        paths.add(getLibName(platform, "System.Globalization.Native" + aotSuffix));
 
-        if (platform.equals("arm64-osx") || platform.equals("x86_64-osx"))
+        if (ExtenderUtil.isMacOSTarget(platform))
         {
-            linkFlags.add(Paths.get(aotBase.toString(), "libRuntime.VxsortEnabled.a").toString());
-
+            paths.add(getLibName(platform, "System.Native"));
+            paths.add(getLibName(platform, "Runtime.VxsortEnabled"));
         }
-        else if (platform.equals("arm64-ios") || platform.equals("x86_64-ios"))
+        else if (ExtenderUtil.isIOSTarget(platform))
         {
-            linkFlags.add(Paths.get(aotBase.toString(), "libstdc++compat.a").toString());
-            linkFlags.add(Paths.get(aotBase.toString(), "libSystem.Net.Security.Native.a").toString());
-            linkFlags.add(Paths.get(aotBase.toString(), "libSystem.Security.Cryptography.Native.Apple.a").toString());
+            paths.add(getLibName(platform, "System.Native"));
+            paths.add(getLibName(platform, "stdc++compat"));
+            paths.add(getLibName(platform, "System.Net.Security.Native"));
+            paths.add(getLibName(platform, "System.Security.Cryptography.Native.Apple"));
             linkFlags.add("-licucore");
         }
+        else if (ExtenderUtil.isWindowsTarget(platform))
+        {
+            paths.add(getLibName(platform, "Runtime.VxsortEnabled"));
+            linkFlags.add("-lbcrypt");
```

Comment:
Maybe it can be moved to `build_input.yml`?

Reply:
Yeah, I guess I'm still trying to figure out what's the most convenient way of doing things.
Now that we have the "platform.sdks.json", we can match sdk and server, we may put more things in the server code (a single place)

Reply:
Also, those libs dependencies come from DotNet which are solely manged by the server.